rhp/v4/rhp.go

```diff
@@ -696,10 +701,11 @@ func RenewalCost(cs consensus.State, p HostPrices, r types.V2FileContractRenewal
 
 // RefreshCost calculates the cost to the host and renter for refreshing a contract.
 func RefreshCost(cs consensus.State, p HostPrices, r types.V2FileContractRenewal, minerFee types.Currency) (renter, host types.Currency) {
+	// Unlike renewals, it is not possible to calculate the renter's cost of the contract during refreshes without the host's prices
+	// because the existing renter and host outputs are rolled into the new contract to ensure the existing data
+	// is still covered by collateral.
 	renter = r.NewContract.RenterOutput.Value.Add(p.ContractPrice).Add(minerFee).Add(cs.V2FileContractTax(r.NewContract)).Sub(r.RenterRollover)
-	// the calculation is different from renewal because the host's revenue is also rolled into the refresh.
-	// This calculates the new collateral the host is expected to put up:
-	// new collateral = (new revenue + existing revenue + new collateral + existing collateral) - new revenue - (existing revenue + existing collateral)
+	// new collateral = (existing revenue + contract price + existing revenue + new collateral + existing collateral) - contract price - (existing revenue + existing collateral)
```

Comment:
`r.NewContract.HostOutput.Value` is the new and existing revenue and collateral plus the contract price right? we have existing revenue twice here 😅